### DISCUSSION TITLE
Question about install nlmixr

### DISCUSSION THREAD

**Discussion by vvn076725282** [nlmixrdevelopment/nlmixr#591] [Q&A]
Hi,officer

Could you please help me? I can't use library(nlmixr). it says

ld: warning: directory not found for option '-L/opt/R/arm64/gfortran/lib/gcc/aarch64-apple-darwin20.2.0/11.0.0'
ld: library not found for -lgfortran
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [nlmixr.so] Error 1
ERROR: compilation failed for package ‘nlmixr’
* removing ‘/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/nlmixr’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/nlmixr’
Warning in install.packages :
  installation of package ‘nlmixr’ had non-zero exit status

The downloaded source packages are in
	‘/private/var/folders/74/vpcrzbw13p119hdy0yclqzgh0000gn/T/RtmpYSmegO/downloaded_packages’

**Comment by vvn076725282** [comment]
Also My laptop is macbook M1

**Comment by vvn076725282** [comment]
And I already download the fortran tool-chain from https://github.com/fxcoudert/gfortran-for-macOS/releases

**Comment by mattfidler** [comment]
Thanks for your patience, 

I do not have any mac computers, but I have a mac runner.  The runner script is here:

https://github.com/nlmixr2/nlmixr2/blob/main/.github/workflows/R-CMD-check.yaml

For this build you have to:
- install homebrew

From homebrew you should install some tools:

```sh
brew install --cask xquartz
brew install cairo
brew install libgit2
brew install udunits
```

Then since you have the fortran tool-chain installed, you will need to install the links to it:

```sh
sudo ln -s /usr/local/bin/gfortran-10 /usr/local/bin/gfortran
sudo mkdir /usr/local/gfortran
sudo ln -s /usr/local/gfortran-10 /usr/local/gfortran 
```

Then you will have to modify your `Makvars` file.  Start R and run the following commands:

```sh
try(dir.create("~/.R"))
writeLines(c("GCCBASE:=$(shell brew --prefix gcc)","FLIBS=-L$(GCCBASE)/lib/gcc/10/lib -lm",""),"~/.R/Makevars")
```

Depending on your architecture, you may need to reinstall `symengine` from source.  First restart the R console and then use the following command:

```R
install.packages("symengine", type="source")
```
After that you should be able to install `RxODE` and `nlmixr` from source.

Since this is a bit complicated, it is easier to use the docker image on the mac. @RichardHooijmaijers knows more about the docker image.
